buildkite/resource_cluster_agent_token.go

```diff
@@ -0,0 +1,193 @@
+package buildkite
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type ClusterAgentToken struct {
+	client *Client
+}
+
+type ClusterAgentTokenResourceModel struct {
+	Id          types.String `tfsdk:"id"`
+	Uuid        types.String `tfsdk:"uuid"`
+	Description types.String `tfsdk:"description"`
+	Token       types.String `tfsdk:"token"`
+	ClusterId   types.String `tfsdk:"cluster_id"`
+	ClusterUuid types.String `tfsdk:"cluster_uuid"`
+}
+
+func NewClusterAgentTokenResource() resource.Resource {
+	return &ClusterAgentToken{}
+}
+
+func (ct *ClusterAgentToken) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cluster_agent_token"
+}
+
+func (ct *ClusterAgentToken) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	ct.client = req.ProviderData.(*Client)
+}
+
+func (ct *ClusterAgentToken) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "A Cluster Agent Token is a token used to connect an agent to a cluster in Buildkite.",
+		Attributes: map[string]resource_schema.Attribute{
+			"id": resource_schema.StringAttribute{
+				Computed: true,
+			},
+			"uuid": resource_schema.StringAttribute{
+				Computed: true,
+			},
+			"description": resource_schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "A description about what this cluster agent token is used for",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
```

Comment:
I don't think this requires replacing the cluster since there is the update mutation. The old agent tokens didnt have that so needed to be replaced

Reply:
@lizrabuya the cluster agent token has an update mutation which allows updating the description of the token in place - it doesn't create a new one. This plan modifier tells the terraform framework that if the user changes the description attribute that it should automatically delete and recreate, which is not necessary. The unclustered agent tokens don't have an update method so needed to be recreated.

So this can be removed from the schema for this attribute.